### ISSUE TITLE
Maybe time to replace CLIP?

### ISSUE THREAD

**Issue by lakshay487** [slavabarkov/tidy#34]
Original CLIP model is very old and no longer SOTA.
I can suggest some fine-tuned CLIP alternatives:
- open clip with the Laion2B dataset https://github.com/mlfoundations/open_clip
- long clip (bigger token lengths during training) https://github.com/beichenzbc/Long-CLIP
- meta-clip https://github.com/facebookresearch/metaclip
- google siglip https://huggingface.co/docs/transformers/main/en/model_doc/siglip different loss function
- apple dfn https://huggingface.co/apple/DFN5B-CLIP-ViT-H-14 if I recall the 5b uses a higher image resolution, amongst a number of other innovations (378 ×378 resolution)
- Blip 1 & 2 by salesforce https://huggingface.co/docs/transformers/en/model_doc/blip

I'm not sure if all of them are of the appropriate size to be able to run on mobile but it's worth taking a look.
